[ORCA-208] Extend all base classes to support Synapse service

BrunoGrandePhD opened the pull request:
The goal of this PR is to implement the service base classes for Synapse. This will make it easier to interact with Synapse in the context of Airflow thanks to the hook class.

Commit:
Create Synapse service

Commit:
Fix typo

Commit:
Close code coverage gap

thomasyu888 commented on src/orca/services/synapse/client_factory.py:
```diff
@@ -0,0 +1,70 @@
+        """
+        profile = client.getUserProfile()
+        if profile.get("userName") == "anonymous":
+            message = "Client is accessing Synapse anonymously."
```
Did we want to support anonymous for publicly accessible files? 

Although, as I type that out, that might only be 20% of the use cases so we can create a ticket to track.

BrunoGrandePhD replied:
Good point, but I agree with your follow-up point. We can create the ticket once someone asks for the feature. #YAGNI

thomasyu888 commented on src/orca/services/synapse/client_factory.py:
```diff
@@ -0,0 +1,70 @@
+        # Ignore authentication error (leave that to `test_client_request`)
+        try:
+            client.login(authToken=auth_token)
+        except SynapseAuthenticationError:
```
Can we check if there are other types of authentication errors thrown by the client?

I'm forgetting if all auth errors extend from this error class.

BrunoGrandePhD replied:
This error came up when I tried creating a client with a fake auth token (`foo`). If others come up here, it should be reported as a bug and then we can decide how to handle it. We might not want to ignore all exceptions here.

thomasyu888 reviewed (approved):
🔥 LGTM! I'm going to pre-approve, just some minor comments.

I haven't had a chance to test, but I trust that it works. I'll install and test when I get a chance.

Commit:
Add basic Synapse integration test

BrunoGrandePhD commented:
I added a basic integration test to make sure that the interaction with Synapse actually works. I added a secret to this repo to enable integration tests in the CI/CD.

Commit:
Pass through Synapse secret
